test/functional/multi-database/multi-database-basic-functionality/multi-database-basic-functionality.ts

```diff
@@ -0,0 +1,151 @@
+        });
+
+        it("should correctly attach and create database files", () => Promise.all(connections.map(async connection => {
+
+            const expectedMainPath = path.join(tempPath, (connections[0].options.database as string).match(/^.*[\\|\/](?<filename>[^\\|\/]+)$/)!.groups!["filename"]);
+
+            expect(fs.existsSync(expectedMainPath)).to.be.true;
+            expect(fs.existsSync(attachAnswerPath)).to.be.true;
+            expect(fs.existsSync(attachCategoryPath)).to.be.true;
+        })));
+
+        it("should prefix tableName when custom database used in Entity decorator", () => Promise.all(connections.map(async connection => {
+
+            const queryRunner = connection.createQueryRunner();
+
+            const tablePathAnswer = `${attachAnswerHandle}.answer`;
+            const table = await queryRunner.getTable(tablePathAnswer);
+            await queryRunner.release();
+
+            const answer = new Answer();
+            answer.text = "Answer #1";
+
+            await connection.getRepository(Answer).save(answer);
+
+            const sql = connection.createQueryBuilder(Answer, "answer")
+                .where("answer.id = :id", {id: 1})
+                .getSql();
+
+            sql.should.be.equal(`SELECT "answer"."id" AS "answer_id", "answer"."text" AS "answer_text" FROM "${attachAnswerHandle}"."answer" "answer" WHERE "answer"."id" = ?`);
+            table!.name.should.be.equal(tablePathAnswer);
+        })));
+
+        it("should not affect tableName when using default main database", () => Promise.all(connections.map(async connection => {
+
+            const queryRunner = connection.createQueryRunner();
+
+            const tablePathUser = `user`;
+            const table = await queryRunner.getTable(tablePathUser);
+            await queryRunner.release();
+
+            const user = new User();
+            user.name = "User #1";
+            await connection.getRepository(User).save(user);
+
+            const sql = connection.createQueryBuilder(User, "user")
+                .where("user.id = :id", {id: 1})
+                .getSql();
+
+            sql.should.be.equal(`SELECT "user"."id" AS "user_id", "user"."name" AS "user_name" FROM "user" "user" WHERE "user"."id" = ?`);
+
+            table!.name.should.be.equal(tablePathUser);
+        })));
+
+        it("should create foreign keys for relations within the same database", () => Promise.all(connections.map(async connection => {
+
+            const queryRunner = connection.createQueryRunner();
+            const tablePathCategory = `${attachCategoryHandle}.category`;
+            const tablePathPost = `${attachCategoryHandle}.post`;
+            const tableCategory = (await queryRunner.getTable(tablePathCategory))!;
+            const tablePost = (await queryRunner.getTable(tablePathPost))!;
+            await queryRunner.release();
+
+            queryRunner.release();
```

Comment:
why you have two `queryRunner.release` calls and why second one doesn't have an await?

Reply:
Removed